runtime/vm/JVMImage.cpp

```diff
@@ -283,28 +288,29 @@ JVMImage::registerEntryInTable(ImageTableHeader *table, UDATA entry)
 }
 
 bool
-JVMImage::readImageFromFile()
+JVMImage::readImageFromFile(void)
 {
 	Trc_VM_ReadImageFromFile_Entry(_heap, _dumpFileName);
 
 	OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
 
 	intptr_t fileDescriptor = omrfile_open(_dumpFileName, EsOpenRead, 0444);
-	if (fileDescriptor == -1) {
+	if (-1 == fileDescriptor) {
 		return false;
 	}
 
-	// Read image header then mmap the rest of the image (heap) into memory
-	JVMImageHeader imageHeader;
-	omrfile_read(fileDescriptor, &imageHeader, sizeof(JVMImageHeader));
+	/* Read image header then mmap the rest of the image (heap) into memory */
+	omrfile_read(fileDescriptor, (void *)_jvmImageHeader, sizeof(JVMImageHeader));
 	uint64_t fileSize = omrfile_flength(fileDescriptor);
-	if (fileSize != sizeof(JVMImageHeader) + imageHeader.imageSize) {
+	if (_jvmImageHeader->imageSize != fileSize) {
 		return false;
 	}
 
-	// Mmap our file into virtual memory
-	JVMImageHeader *block = (JVMImageHeader*)mmap((void*)imageHeader.heapAddress, sizeof(JVMImageHeader) + imageHeader.imageSize, PROT_READ, MAP_PRIVATE, fileDescriptor, 0);
-	_heap = (J9Heap*)(block + 1);
+	_jvmImageHeader = (JVMImageHeader *)mmap(
+		(void *)_jvmImageHeader->heapAddress,
+		sizeof(JVMImageHeader) + _jvmImageHeader->imageSize,
```

Comment:
here too use imageSize instead of `sizeof(JVMImageHeader) + _jvmImageHeader->imageSize`